### PR TITLE
feat(Rust Lang Book): add presence

### PR DESCRIPTION
**Pull request by eliyya**
## Description 

presence for the official rust language book

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `npm run format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image

</details>

### CODE REVIEW

File: websites/R/Rust Lang Book/metadata.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"id": "534614025271771158",
+		"name": "eli_yya#3893"
```
Comment (theusaf) [inline websites/R/Rust Lang Book/metadata.json:5]: Discriminator added mistakenly by outdated `pmd` program.
```suggestion
		"name": "eli_yya"
```

File: websites/R/Rust Lang Book/presence.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const presence = new Presence({
+	clientId: "1033608073106968576",
+});
+
+presence.on("UpdateData", async () => {
+	const path = window.location.pathname
+			.replace("/book/", "")
+			.replace(".html", ""),
+		presenceData: PresenceData = {
+			largeImageKey: "rust-logo-512",
```
Comment (theusaf) [inline websites/R/Rust Lang Book/presence.ts:10]: Imgur links are preferred over Discord asset names. This applies to `smallImageKey` as well.
```suggestion
			largeImageKey: "https://i.imgur.com/kTnc40Q.png",
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Comment by eliyya** [comment]
> ## Description
> presence for the official rust language book
> 
> ## Acknowledgements
> * [x]  I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
> * [x]  I linted the code by running `npm run format`
> * [x]  The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)
> 
> ## Screenshots
> Proof showing the creation/modification is working as expected

I noticed that for some reason the screenshots are not shown in the original message so I am adding them here

image
image
image

File: websites/R/Rust Lang Book/presence.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	const path = window.location.pathname
+			.replace("/book/", "")
+			.replace(".html", ""),
+		presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/kTnc40Q.png", // rust logo 512x512
+			smallImageKey: "fhttps://i.imgur.com/B7Rip9I.png", // ferris 512x512
+			smallImageText: "Hi rustaceans!",
+			buttons: [
+				{
+					label: "Read the book",
```
Comment (EncryptedDev) [inline websites/R/Rust Lang Book/presence.ts:15]: Shorter buttons are preferable
```suggestion
					label: "Read Book",
```

File: websites/R/Rust Lang Book/presence.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+});
+
+presence.on("UpdateData", async () => {
+	const path = window.location.pathname
+			.replace("/book/", "")
+			.replace(".html", ""),
+		presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/kTnc40Q.png", // rust logo 512x512
+			smallImageKey: "fhttps://i.imgur.com/B7Rip9I.png", // ferris 512x512
+			smallImageText: "Hi rustaceans!",
```
Comment (EncryptedDev) [inline websites/R/Rust Lang Book/presence.ts:12]: Not needed

File: websites/R/Rust Lang Book/presence.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+	clientId: "1033608073106968576",
+});
+
+presence.on("UpdateData", async () => {
+	const path = window.location.pathname
+			.replace("/book/", "")
+			.replace(".html", ""),
+		presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/kTnc40Q.png", // rust logo 512x512
+			smallImageKey: "fhttps://i.imgur.com/B7Rip9I.png", // ferris 512x512
```
Comment (EncryptedDev) [inline websites/R/Rust Lang Book/presence.ts:11]:
```suggestion
			smallImageKey: "https://i.imgur.com/B7Rip9I.png", // ferris 512x512
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/R/Rust Lang Book/presence.ts (hunk 4 above)
Comment (eliyya) [reply websites/R/Rust Lang Book/presence.ts:12]: the `smallImageKey` in url form is not needed?
I return it to the way it was?

Comment (EncryptedDev) [reply websites/R/Rust Lang Book/presence.ts:12]: goofy diff

I’m only referring to L12

Comment (eliyya) [reply websites/R/Rust Lang Book/presence.ts:12]: Is it necessary to remove it? I would prefer to keep it, it is a small detail.

Comment (EncryptedDev) [reply websites/R/Rust Lang Book/presence.ts:12]: It is necessary to remove it, yes

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**